token-lending/js/src/instructions/borrowObligationLiquidity.ts

```diff
@@ -0,0 +1,58 @@
+import { TOKEN_PROGRAM_ID } from '@solana/spl-token';
+import { PublicKey, SYSVAR_CLOCK_PUBKEY, TransactionInstruction } from '@solana/web3.js';
+import BN from 'bn.js';
+import { struct, u8 } from 'buffer-layout';
+import { LENDING_PROGRAM_ID } from '../constants';
+import { u64 } from '../util';
+import { LendingInstruction } from './instruction';
+
+export const borrowObligationLiquidityInstruction = (
+    liquidityAmount: number | BN,
+    sourceLiquidity: PublicKey,
+    destinationLiquidity: PublicKey,
+    borrowReserve: PublicKey,
+    borrowReserveLiquidityFeeReceiver: PublicKey,
+    obligation: PublicKey,
+    lendingMarket: PublicKey,
+    lendingMarketAuthority: PublicKey,
+    obligationOwner: PublicKey,
+    hostFeeReceiver?: PublicKey
+): TransactionInstruction => {
+    const dataLayout = struct([u8('instruction'), u64('liquidityAmount')]);
+
+    const data = Buffer.alloc(dataLayout.span);
```

Comment:
This requires the client to either be a node app or have a bundler which can polyfill this global. Have you had any issues with using it in oyster-lending?

Reply:
Oyster uses create-react-app which handles the bundling (I think using Webpack) so that takes care of the polyfill. We could us Rollup for this library and build Node and browser versions, but I don't mind keeping this library simple and leaving the choice to the app that depends on it. I'll make a note in the readme for now, and add a bundler to the package if we get feedback.

Reply:
I looked into this and Webpack 5 recently removed support for automatically polyfilling this, which makes it more annoying if you're using this as expected (usually for the web). I didn't want to add a bundler, but I'd rather do that than make users deal with it.

Reply:
Added rollup + config. Will test using Oyster Lending and follow up in a new PR.